gsl/span

```diff
@@ -630,7 +630,7 @@ template <class ElementType, std::ptrdiff_t Extent>
 constexpr ElementType& at(const span<ElementType, Extent>& s, size_t index)
 {
     // No bounds checking here because it is done in span::operator[] called below
-    return s[index];
+    return s[narrow_cast<typename span<ElementType, Extent>::index_type>(index)];
```

Comment:
This is the only one that I was not sure about what to do as we could use `gsl::narrow` or `gsl::narrow_cast`. I used `gsl::narrow_cast` as I know a lot of people have brought up issues with performance so I figured that would be the likely choice.